source/common/ssl/context_impl.cc

```diff
@@ -617,6 +617,12 @@ ServerContextImpl::ServerContextImpl(Stats::Scope& scope, const ServerContextCon
                    SSL_CTX_get_app_data(SSL_get_SSL_CTX(client_hello->ssl)))
             ->selectTlsContext(client_hello);
       });
+  // Compute the session context ID hash. We use all the certificate identities,
+  // since we should have a common ID for session resumption no matter what cert
+  // is used.
+  uint8_t session_context_buf[EVP_MAX_MD_SIZE] = {};
+  unsigned session_context_len = 0;
+  hashSessionContextId(server_names, session_context_buf, session_context_len);
```

Comment:
Nit: perhaps `generateHashForSessionContexId`? (it's more about `For` than `generate`, but both would be good).